File: solved/all-paths-from-source-to-target.go
func allPathsSourceTarget(graph [][]int) [][]int {
    var res [][]int
    var resolve func (path []int, cur int)
    resolve = func (path []int, cur int) {
        if cur == len(graph) - 1 {
            res = append(res, append([]int(nil), path...))
        } else {
            for _, next := range graph[cur] {
                resolve(append(path, next), next)
            }
        }
    }
    resolve([]int{0}, 0)
    return res
}

File: solved/all-paths-from-source-to-target.rs
impl Solution {
    pub fn resolve(g: &Vec<Vec<i32>>, output: &mut Vec<Vec<i32>>, path: &mut Vec<i32>, cur: usize) {
        if cur == g.len() - 1 {
            output.push(path.clone());
        } else {
            g[cur].iter().for_each(|&next| {
                path.push(next);
                Self::resolve(g, output, path, next as usize);
                path.pop();
            });
        }
    }

    pub fn all_paths_source_target(graph: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let mut res = Vec::new();
        Self::resolve(&graph, &mut res, &mut vec![0], 0);
        res
    }
}
